setup.py

```diff
@@ -1,6 +1,21 @@
+import setuptools
 from setuptools import setup
 from versioningit import get_cmdclasses
 
+if int(setuptools.__version__.split(".")[0]) < 61:
+    raise RuntimeError(
+        "At least setuptools 61 is required to install qcodes from source"
```

Comment:
do i understand correctly that we need these guards for cases where `pip install ...` is not used, which is not a commonplace but nice to keep being supported?

Reply:
I think there are 2 ways we could get here with a version of setuptools that is too old. 

* Installing from source using either `python setup.py install` or `pip install . --no-build-isolation` from an env that has an older version of setuptools or using some other tool that does not respect pep 517 build isolation. 
* Using a version of pip that is too old to support pep517 (this was added in pip 19)

In any case this will only happen if you install from source not from a wheel)

Reply:
then we are quite safe :)

Reply:
verified manually that editable install using pip 19.0.0 works as expected